eazieda/outliers_detect.py

```diff
@@ -0,0 +1,167 @@
+import numpy as np
+
+from scipy import stats
+from sklearn.ensemble import IsolationForest
+
+
+def outliers_detect(s, method="zscore", remove=False):
+    """
+    Detects outliers in a pandas series
+
+    Parameters
+    ----------
+    s : pandas.core.series.Series
+        Pandas Series for which the outliers need to be found
+
+    method : str, default = "zscore"
+        The algorithm/method used for outlier detection.
+        One of 'zscore',  'iforest', 'iqr'
+
+    remove : bool, default = False
+        in-place removal of the outliers
+
+    Returns
+    -------
+    numpy.array
+        Boolean array with same length as the input,
+        indices of outlier marked.
+
+    Examples
+    --------
+    >>> from eazieda.outliers_detect import outliers_detect_zscore
+    >>> s = pd.Series([1,1,1,1,1,1,1,1,1,1,1e14])
+    >>> outliers_detect_zscore(s)
+    array([False, False, False, False, False, False, False, False, False,
+        True])
+    """
+    if method == "zscore":
+        outliers = outliers_detect_zscore(s)
+    elif method == "iqr":
+        outliers = outliers_detect_iqr(s)
+    elif method == "iforest":
+        outliers = outliers_detect_iforest(s)
+
+    if remove:
+        remove_outliers(s, outliers)
+
+    return outliers
+
+
+def outliers_detect_iforest(s):
+    """
+    Detects outliers in a pandas series using isolation forests
+
+    Parameters
+    ----------
+    s : pandas.core.series.Series
+        Pandas Series for which the outliers need to be found
+
+    Returns
+    -------
+    numpy.array
+        Boolean array with same length as the input,
+        indices of outlier marked.
+
+    Examples
+    --------
+    >>> from eazieda.outliers_detect import outliers_detect_iqr
+    >>> s = pd.Series([1,2,1,2,1, 1000])
+    >>> outliers_detect_iforest(s)
+    array([False, False, False, False, False,  True])
+    """
+    iforest = IsolationForest().fit(s.values.reshape(-1, 1))
+    return iforest.predict(s.values.reshape(-1, 1)) == -1
+
+
+def outliers_detect_iqr(s, factor=1.5):
+    """
+    Detects outliers in a pandas series using inter-quantile ranges
+
+    Parameters
+    ----------
+    s : pandas.core.series.Series
+        Pandas Series for which the outliers need to be found
+
+    factor : int
+        iqr factor used for outliers
+
+    Returns
+    -------
+    numpy.array
+        Boolean array with same length as the input,
+        indices of outlier marked.
+
+    Examples
+    --------
+    >>> from eazieda.outliers_detect import outliers_detect_iqr
+    >>> s = pd.Series([1,2,1,2,1, 1000])
+    >>> outliers_detect_zscore(s)
+    array([False, False, False, False, False,  True])
+    """
+    q1 = s.quantile(0.25)
+    q3 = s.quantile(0.75)
+    inter_quantile_range = q3 - q1
+    return (
+        (s < (q1 - 1.5 * inter_quantile_range)) |
+        (s > (q3 + 1.5 * inter_quantile_range))
```

Comment:
I think you need to add in factor here?
```suggestion
        (s < (q1 - factor* inter_quantile_range)) |
        (s > (q3 + factor* inter_quantile_range))
```